docs/user/implement_env.md

```diff
@@ -0,0 +1,308 @@
+# Adding a New Environment
+
+Garage uses the [`Environment`](https://github.com/rlworkgroup/garage/blob/master/src/garage/_environment.py)
+API for all RL environments.
+
+The public API methods of :`Environment` are:
+
+* `reset()`: Resets the environment. Return the first observation, and the episode-level information.
+* `step()`: Steps the environment with an action and returns an EnvStep.
+
+    An `EnvStep` is a named tuple of `env_spec`, `observation`, `action`, `reward`,
+    `step_type` representing a single step in the environment.
+* `render()`: Returns value for rendering, could be a terminal text, an image, etc.
+* `visualize()`: Creates a visualization of the environment.
+* `close()`: Closes the environment.
+
+The attributes of `Environment` are:
+
+* `action_space`: The action space specification.
+* `observation_space`: The observation space specification.
+* `spec`: The environment specifications.
+* `render_modes`: The list of supported render modes.
+
+You should expect to use all environments in garage with this API.
+
+## Add Support for an Existing Environment Library
+
+Luckily, Garage already came with a variety of external environment wrappers:
+
+* `garage.envs.GymEnv`, which supports the `gym` API.
```

Comment:
I think class names are sufficient here rather than fully-qualified module paths. You can refer to these using ``` :class:`GymEnv` ``` (for instance) to link to the API reference.

Reply:
I don't think sphnix will create a link to the API in markdown file with :class:`GymEnv`. Please check the read the docs result.

Seems like this only works in docstrings.